src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/model/Policy.kt

```diff
@@ -155,7 +162,22 @@ data class Policy(
                             states.add(State.parse(xcp))
                         }
                     }
-                    ISM_TEMPLATE -> ismTemplate = if (xcp.currentToken() == Token.VALUE_NULL) null else ISMTemplate.parse(xcp)
+                    ISM_TEMPLATE -> {
+                        if (xcp.currentToken() != Token.VALUE_NULL) {
+                            ismTemplates = mutableListOf()
+                            when (xcp.currentToken()) {
+                                Token.START_ARRAY -> {
+                                    while (xcp.nextToken() != Token.END_ARRAY) {
+                                        ismTemplates.add(ISMTemplate.parse(xcp))
+                                    }
+                                }
+                                Token.START_OBJECT -> {
+                                    ismTemplates.add(ISMTemplate.parse(xcp))
+                                }
+                                else -> ensureExpectedToken(Token.START_ARRAY, xcp.currentToken(), xcp)
```

Comment:
Hmm what's the point of this else branch?

Reply:
just for sanity, when user input not array, not object, showing an exception telling them you should input array